scene/resources/animation.cpp

```diff
@@ -5476,469 +5476,622 @@ bool Animation::_fetch_compressed_by_index(uint32_t p_compressed_track, int p_in
-			return Rect2i(int32_t(ra.position.x + rb.position.x * c + 0.5), int32_t(ra.position.y + rb.position.y * c + 0.5), int32_t(ra.size.x + rb.size.x * c + 0.5), int32_t(ra.size.y + rb.size.y * c + 0.5));
-		}
+		} break;
 		case Variant::VECTOR3: {
 			return (a.operator Vector3()) + (b.operator Vector3()) * c;
-		}
-		case Variant::VECTOR3I: {
-			const Vector3i va = a.operator Vector3i();
-			const Vector3i vb = b.operator Vector3i();
-			return Vector3i(int32_t(va.x + vb.x * c + 0.5), int32_t(va.y + vb.y * c + 0.5), int32_t(va.z + vb.z * c + 0.5));
-		}
+		} break;
 		case Variant::VECTOR4: {
 			return (a.operator Vector4()) + (b.operator Vector4()) * c;
-		}
-		case Variant::VECTOR4I: {
-			const Vector4i va = a.operator Vector4i();
-			const Vector4i vb = b.operator Vector4i();
-			return Vector4i(int32_t(va.x + vb.x * c + 0.5), int32_t(va.y + vb.y * c + 0.5), int32_t(va.z + vb.z * c + 0.5), int32_t(va.w + vb.w * c + 0.5));
-		}
+		} break;
 		case Variant::PLANE: {
 			const Plane pa = a.operator Plane();
 			const Plane pb = b.operator Plane();
 			return Plane(pa.normal + pb.normal * c, pa.d + pb.d * c);
-		}
+		} break;
 		case Variant::COLOR: {
 			return (a.operator Color()) + (b.operator Color()) * c;
-		}
+		} break;
 		case Variant::AABB: {
 			const ::AABB aa = a.operator ::AABB();
 			const ::AABB ab = b.operator ::AABB();
 			return ::AABB(aa.position + ab.position * c, aa.size + ab.size * c);
-		}
+		} break;
 		case Variant::BASIS: {
 			return (a.operator Basis()) + (b.operator Basis()) * c;
-		}
+		} break;
 		case Variant::QUATERNION: {
 			return (a.operator Quaternion()) * Quaternion().slerp((b.operator Quaternion()), c);
-		}
+		} break;
 		case Variant::TRANSFORM2D: {
 			return (a.operator Transform2D()) * Transform2D().interpolate_with((b.operator Transform2D()), c);
-		}
+		} break;
 		case Variant::TRANSFORM3D: {
 			return (a.operator Transform3D()) * Transform3D().interpolate_with((b.operator Transform3D()), c);
-		}
+		} break;
+		case Variant::BOOL:
+		case Variant::INT:
+		case Variant::RECT2I:
+		case Variant::VECTOR2I:
+		case Variant::VECTOR3I:
+		case Variant::VECTOR4I:
+		case Variant::PACKED_INT32_ARRAY:
+		case Variant::PACKED_INT64_ARRAY: {
+			// Fallback the interpolatable value which needs casting.
+			return cast_from_blendwise(blend_variant(cast_to_blendwise(a), cast_to_blendwise(b), c), a.get_type());
+		} break;
+		case Variant::STRING:
+		case Variant::STRING_NAME: {
+			Array arr_a = cast_to_blendwise(a);
+			Array arr_b = cast_to_blendwise(b);
+			int min_size = arr_a.size();
+			int max_size = arr_b.size();
+			bool is_a_larger = inform_variant_array(min_size, max_size);
+			int mid_size = interpolate_variant(arr_a.size(), arr_b.size(), c);
+			if (is_a_larger) {
+				arr_a.resize(mid_size);
+			} else {
+				arr_b.resize(mid_size);
+			}
+			return cast_from_blendwise(blend_variant(arr_a, arr_b, c), a.get_type());
+		} break;
+		case Variant::PACKED_BYTE_ARRAY: {
+			// Skip.
+		} break;
```

Comment:
At last, I just added exception handling for PackedByteArray, although I think that's a very rare case where it would be needed.